themes/doom-gruvbox-theme.el

```diff
@@ -20,66 +20,93 @@ determine the exact padding."
 
   ;; name        gui       256       16
-  ((bg         '("#282828" "#282828"  nil         ))
-   (bg-alt     '("#323232" "#323232"  nil         ))
-   (accent     '("#504945" "#504945" "brown"      ))
-
-   (base0      '("#1B2229" "black"   "black"      ))
-   (base1      '("#151617" "#101010" "brightblack"))
-   (base2      '("#1d1f20" "#191919" "brightblack"))
-   (base3      '("#2d2e2e" "#252525" "brightblack"))
-   (base4      '("#4e4e4e" "#454545" "brightblack"))
-   (base5      '("#555556" "#6b6b6b" "brightblack"))
-   (base6      '("#7c6f64" "#7b7b7b" "brightblack"))
-   (base7      '("#cfc0c5" "#c1c1c1" "brightblack"))
-   (base8      '("#ffffff" "#ffffff" "brightwhite"))
-   (fg         '("#ebdbb2" "#dfdfdf" "brightwhite"))
-   (fg-alt     '("#928374" "#dfdfdf" "brightwhite"))
-
-   (grey       '("#555556" "#515154" "brightblack"))
-   (red        '("#fb4934" "#e74c3c" "red"))
-   (magenta    '("#fb2874" "#fb2874" "magenta"))
-   (violet     '("#d3869b" "#d3869b" "brightmagenta"))
-   (orange     '("#fe8019" "#fd971f" "brightred"))
-   (yellow     '("#fabd2f" "#fabd2f" "yellow"))
-   (dark-green '("#689d6a" "#689d6a" "green"))
-   (green      '("#8ec07c" "#8ec07c" "green"))
-   (teal       green)
-   (olive      '("#b8bb26" "#b8bb26" "green"))
-   (blue       '("#268bd2" "#2686D6" "brightblue"))
-   (dark-blue  '("#727280" "#727280" "blue"))
-   (cyan       '("#83a598" "#83a598" "brightcyan"))
-   (dark-cyan  '("#458588" "#458588" "cyan"))
+  ((bg
+    (cond ((equal doom-gruvbox-dark-variant "hard") '("#1d2021" "#1e1e1e" nil))   ; bg0_h
+          ((equal doom-gruvbox-dark-variant "soft") '("#32302f" "#323232" nil))   ; bg0_s
+          (t                                        '("#282828" "#282828" nil)))) ; bg0
+   (bg-alt
+    (cond ((equal doom-gruvbox-dark-variant "hard") '("#282828" "#282828" nil))
+          ((equal doom-gruvbox-dark-variant "soft") '("#403d3d" "#404040" nil))
+          (t                                        '("#32302f" "#323232" nil))))
+   (bg-alt2    '("#504945" "#504945" "brown"      )) ; bg2 (for region, selection etc.)
+
+   (base0      '("#0d1011" "black"   "black"      )) ; (self-defined)
+   (base1      '("#1d2021" "#1d1d1d" "brightblack")) ; bg0_h
+   (base2      '("#282828" "#282828" "brightblack")) ; bg0
+   (base3      '("#3c3836" "#383838" "brightblack")) ; bg1
+   (base4      '("#665c54" "#5c5c5c" "brightblack")) ; bg3
+   (base5      '("#7c6f64" "#6f6f6f" "brightblack")) ; bg4
+   (base6      '("#928374" "#909090" "brightblack")) ; gray
+   (base7      '("#d5c4a1" "#cccccc" "brightblack")) ; fg2
+   (base8      '("#fbf1c7" "#fbfbfb" "brightwhite")) ; fg0
+   (fg         '("#ebdbb2" "#dfdfdf" "brightwhite")) ; fg/fg1
+   (fg-alt     '("#d5c4a1" "#cccccc" "brightwhite")) ; fg2
+
+   ;; Standardized official colours from gruvbox
+   (grey       '("#928374" "#909090" "brightblack"))   ; gray
+   (gray       '("#928374" "#909090" "brightblack"))   ; gray
```

Comment:
Can you remove `gray`? Not sure why we need a duplicate; it isn't even used.

Reply:
Sure, removed.